optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -0,0 +1,172 @@
+import static io.camunda.optimize.util.ZeebeBpmnModels.START_EVENT;
+import static io.camunda.optimize.util.ZeebeBpmnModels.USER_TASK;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createProcessWith83SignalEvents;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleServiceTaskProcess;
+import static io.camunda.optimize.util.ZeebeBpmnModels.createSimpleUserTaskProcess;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.optimize.dto.optimize.FlowNodeDataDto;
+import io.camunda.zeebe.model.bpmn.Bpmn;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.assertj.core.groups.Tuple;
+import org.camunda.bpm.model.bpmn.BpmnModelInstance;
+import org.camunda.bpm.model.bpmn.instance.EndEvent;
+import org.camunda.bpm.model.bpmn.instance.IntermediateCatchEvent;
+import org.camunda.bpm.model.bpmn.instance.Signal;
+import org.camunda.bpm.model.bpmn.instance.StartEvent;
+import org.camunda.bpm.model.xml.ModelInstance;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.junit.jupiter.api.Test;
+
+public class BpmnModelUtilTest {
+
+  public static final String CUSTOMER_ONBOARDING = "CustomerOnboarding";
+  public static final String END_EVENT_TYPE = "endEvent";
+  public static final String SERVICE_TASK_TYPE = "serviceTask";
+
+  public static final String SIMPLE_SERVICE_TASK_PROCESS =
+      Bpmn.convertToString(createSimpleServiceTaskProcess(CUSTOMER_ONBOARDING));
+
+  public static final String PROCESS_WITH_83_SIGNAL_EVENTS =
+      Bpmn.convertToString(createProcessWith83SignalEvents(CUSTOMER_ONBOARDING));
+
+  @Test
+  void shouldParseBpmnModel() {
+    // when
+    final BpmnModelInstance modelInstance = parseBpmnModel(SIMPLE_SERVICE_TASK_PROCESS);
+
+    // then
+    assertThat(modelInstance).isNotNull();
+  }
+
+  @Test
+  void shouldExtractFlowNodeData() {
+    // when
+    final List<FlowNodeDataDto> flowNodeData =
+        BpmnModelUtil.extractFlowNodeData(SIMPLE_SERVICE_TASK_PROCESS);
+
+    // then
+    assertThat(flowNodeData)
+        .isNotNull()
+        .isNotEmpty()
+        .extracting(FlowNodeDataDto::getId, FlowNodeDataDto::getType, FlowNodeDataDto::getName)
+        .containsExactlyInAnyOrder(
+            Tuple.tuple(START_EVENT, START_EVENT, START_EVENT),
+            Tuple.tuple(SERVICE_TASK, SERVICE_TASK_TYPE, SERVICE_TASK),
+            Tuple.tuple(END_EVENT, END_EVENT_TYPE, null));
+  }
+
+  @Test
+  void shouldExtractUserTaskNames() {
+    // when
+    final String bpmnModelInstance =
+        Bpmn.convertToString(createSimpleUserTaskProcess(CUSTOMER_ONBOARDING));
+    final Map<String, String> userTaskNames = BpmnModelUtil.extractUserTaskNames(bpmnModelInstance);
+
+    // then
+    assertThat(userTaskNames).isNotNull().isNotEmpty().containsKey(USER_TASK);
+  }
+
+  @Test
+  void shouldExtractProcessDefinitionName() {
```

Comment:
I think there is a missing test for the scenario where no definition name is present for the given key. For example:
```
BpmnModelUtil.extractProcessDefinitionName(
            "someUnknownKey", SIMPLE_SERVICE_TASK_PROCESS)
```
This should return an empty optional. Let's have a test to verify this

Reply:
fixed